core/domain/platform_feature_services_test.py

```diff
@@ -499,3 +499,34 @@ def test_platform_parameter_schema_acc_to_data_type(self) -> None:
             {'type': 'float'},
             feature_services.get_platform_parameter_schema(self.param_c.name)
         )
+
+    def test_raise_exception_when_invalid_data_type_trying_to_get_schema(
+        self) -> None:
+        param_dict = {
+            'name': 'param_name',
+            'description': 'param description',
+            'data_type': 'unknown',
+            'rules': [],
+            'rule_schema_version': (
+                feconf.CURRENT_PLATFORM_PARAMETER_RULE_SCHEMA_VERSION),
+            'default_value': 'abc',
+            'is_feature': False,
+            'feature_stage': None
+        }
+        # Here we use MyPy ignore because we want to create a platform parameter
+        # with an invalid 'data_type' field to test that the exception
+        # gets raised.
+        parameter = platform_parameter_domain.PlatformParameter.from_dict(
+            param_dict) # type: ignore[arg-type]
+        swap_get_platform_parameter = self.swap_to_always_return(
+            registry.Registry,
+            'get_platform_parameter',
+            parameter
+        )
+
+        with swap_get_platform_parameter, self.assertRaisesRegex(Exception, (
+            'The %s platform parameter does not have a valid '
+            'data type, must be one of %s inorder to get schema.' % (
```

Comment:
Do not interpolate here, test against the exact error message. Do not use logic in tests.

Reply:
Done